### PR TITLE
Duplicate item

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/utils/rest-api.js [hunk 1]
```diff
@@ -384,6 +409,30 @@ export function createCase({ name, description, file, parentDirectoryUuid }) {
     }).then((response) => handleResponse(response, false));
 }
 
+export function duplicateCase(
+    name,
+    description,
+    parentCaseUuid,
+    parentDirectoryUuid
+) {
+    console.info('Creating a new case...');
```
Comment (FranckLecuyer) [inline src/utils/rest-api.js]: Typo : Duplicating a case

File: src/utils/rest-api.js [hunk 2]
```diff
@@ -362,6 +362,31 @@ export function createStudy(
...
+    studyDescription,
+    parentStudyUuid,
+    parentDirectoryUuid
+) {
+    console.info('Creating a new study...');
+    let urlSearchParams = new URLSearchParams();
+    urlSearchParams.append('duplicateFrom', parentStudyUuid);
+    urlSearchParams.append('studyName', studyName);
+    urlSearchParams.append('description', studyDescription);
+    urlSearchParams.append('parentStudyUuid', parentStudyUuid);
```
Comment (FranckLecuyer) [inline src/utils/rest-api.js]: parameter not used in explore server

File: src/components/menus/content-contextual-menu.js [hunk 3]
```diff
@@ -86,6 +93,103 @@ const ContentContextualMenu = (props) => {
         setOpenDialog(dialogId);
     };
 
+    const duplicateItem = () => {
+        if (activeElement) {
+            const duplicateSuffix = '(1)';
+            elementExists(
```
Comment (FranckLecuyer) [inline src/components/menus/content-contextual-menu.js:99]: Maybe later, add an api in directory server to generate a new unique name, and use it here ...

File: src/components/menus/content-contextual-menu.js [hunk 4]
```diff
@@ -86,6 +93,103 @@ const ContentContextualMenu = (props) => {
...
+                selectedDirectory.elementUuid,
+                activeElement.elementName + duplicateSuffix,
```
Comment (FranckLecuyer) [inline src/components/menus/content-contextual-menu.js]: Maybe use a local variable, set to : activeElement.elementName + duplicateSuffix
and use it elsewhere

File: src/components/menus/content-contextual-menu.js [hunk 5]
```diff
@@ -86,6 +93,103 @@ const ContentContextualMenu = (props) => {
...
+                        case ElementType.CASE:
+                            duplicateCase(
+                                activeElement.elementName + duplicateSuffix,
+                                activeElement.description,
+                                activeElement.elementUuid,
+                                selectedDirectory.elementUuid
+                            )
+                                .then(() => {
+                                    onClose();
+                                    setHideMenu(false);
```
Comment (FranckLecuyer) [inline src/components/menus/content-contextual-menu.js]: Maybe handleCloseDialog() ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by AbdelHedhili — APPROVED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Pull request merged**